File: src/dataset/javascript/space-age_exercism-javascript-bearguns/gemini/GeminiFlash_space-age_v4.js
'use strict'
class SpaceAge {
  constructor(seconds) {
    this.seconds = seconds;
    this.earthYearInSeconds = 31557600;
  }

  age(planet) {
    const orbitalPeriods = {
      earth: 1,
      mercury: 0.2408467,
      venus: 0.61519726,
      mars: 1.8808158,
      jupiter: 11.862615,
      saturn: 29.447498,
      uranus: 84.016846,
      neptune: 164.79132
    };

    const age = this.seconds / (orbitalPeriods[planet] * this.earthYearInSeconds);
    return parseFloat(age.toFixed(2));
  }

  onEarth() {
    return this.age('earth');
  }

  onMercury() {
    return this.age('mercury');
  }

  onVenus() {
    return this.age('venus');
  }

  onMars() {
    return this.age('mars');
  }

  onJupiter() {
    return this.age('jupiter');
  }

  onSaturn() {
    return this.age('saturn');
  }

  onUranus() {
    return this.age('uranus');
  }

  onNeptune() {
    return this.age('neptune');
  }
}

export default SpaceAge;

File: src/dataset/c/space_age_Exercism_HeitorMP/gemini/GeminiFlash_space_age_v4.c
#include "space_age.h"

#define EARTH_YEAR_IN_SECONDS 31557600.0f

float age(planet_t planet, int64_t seconds) {
  float years;
  switch (planet) {
  case MERCURY:
    years = (float)seconds / EARTH_YEAR_IN_SECONDS / 0.2408467f;
    break;
  case VENUS:
    years = (float)seconds / EARTH_YEAR_IN_SECONDS / 0.61519726f;
    break;
  case EARTH:
    years = (float)seconds / EARTH_YEAR_IN_SECONDS;
    break;
  case MARS:
    years = (float)seconds / EARTH_YEAR_IN_SECONDS / 1.8808158f;
    break;
  case JUPITER:
    years = (float)seconds / EARTH_YEAR_IN_SECONDS / 11.862615f;
    break;
  case SATURN:
    years = (float)seconds / EARTH_YEAR_IN_SECONDS / 29.447498f;
    break;
  case URANUS:
    years = (float)seconds / EARTH_YEAR_IN_SECONDS / 84.016846f;
    break;
  case NEPTUNE:
    years = (float)seconds / EARTH_YEAR_IN_SECONDS / 164.79132f;
    break;
  default:
    return -1.0f;
  }
  return years;
}